### ISSUE TITLE
Make possible to encode data in the app

### ISSUE THREAD

**Issue by User1l0** [sz3/cfc#17]
Using the internet to encode your data really makes this app unusable. I'll be very pleased if this app could fully function at it's own

**Comment by sz3** [comment]
Ok, let me try again:

If you download the [cimbar.html](https://github.com/sz3/libcimbar/releases/download/v0.5.12/cimbar.html) file to your phone, you can open it with the files app (or equivalent) and use the web browser (even offline!) to encode.

Ideally cfc would be able to do this too, but the reality is that this is a hobby project right now and there are other parts of it (improving the format, improving the color decode) that I'm focusing on when I get a chance to work on it.

**Comment by nathgit** [comment]
I downloaded the html file & it's blank. I tried opening it with Chrome browser & 2 different text editors. In all cases it shows it's blank. No offense, but this isn't useful for me at all, without the ability to transfer things totally offline on both sides. If I didn't care about it working offline, I already have a way to transfer files over the internet. I was only interested in this because it claimed it worked offline.

**Comment by justinormont** [comment]
@sz3: What do you think about adding an Android WebView within the app, which displays a locally stored copy of the website?

**Comment by sz3** [comment]
> @sz3: What do you think about adding an Android WebView within the app, which displays a locally stored copy of the website?

I think it's a good idea. (in particular, re-using the web encoding should make it easier to maintain)

**Comment by sz3** [comment]
> > @sz3: What do you think about adding an Android WebView within the app, which displays a locally stored copy of the website?
> 
> What about just trying to include this feature in the app. It will save your time while using the app. No offense but your app Is not complete at all...

They are suggesting a way to include the feature in the app, but in a way that will hopefully not create much burden on the maintainer (me!). Which is appreciated.

For what it's worth, I think this is mostly a communication issue:
1. The cimbar.js encoder *does work* offline. You don't need an internet connection to run the "website", it's just packaged as a website. But, I didn't realize (unfortunately, given my answer above from 6 months ago) that many versions of Android now prevent users from opening html files directly. It didn't used to be that way, which makes my previous advice out of date... But that's also why...
2. The cimbar.org website (which is a hosted cimbar.js) is now a progressive web app (PWA) that can be installed and run offline. This would be the preferred way to run the encoder on android offline.

I never mentioned (2) in this thread, so that's my bad.

**Comment by deknos** [comment]
Hi,
yes, i second this, please make the app in a way, however you do it, that you can select files from the smartphone and encode them, so another smartphone with your app can decode it and download. It's okay, if it's an webappview, i just want to install your app on two phones and have the encode and decode functionality which works offline without internet or wireless, only with camera and screens.

I would even pitch some money in some money if you have a liberay/patreon account.

**Comment by mrwm** [comment]
I did a quick webview test, but quickly found out that getting the file picker to open on a webview is quite a task 👀 

On the other hand, it's not all bad news. The webview works offline, assuming the user opened the webview to load https://cimbar.org/ in the first launch for the webview to save it to cache.

**Comment by mrwm** [comment]
Here's a screen record of the webview launched after a button press. I've recorded to show that the webview loads in online mode, and offline mode.

https://github.com/user-attachments/assets/59516891-a355-4bb0-b671-b5f63fffffbe

---

The only unfortunate part is that I couldn't figure out how to get the file picker to work with webview 🥲
